### PR TITLE
lxd_snapshot resource

### PR DESCRIPTION
**Pull request by sl1pm4t**
This allows creating, destroying container snapshots.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -282,6 +282,17 @@ resource "lxd_volume" "volume1" {
...
+
+The `lxd_snapshot` resource can be used to create a point in time snapshot of a container.
+
+```hcl
+resource "lxd_snapshot" "snap1" {
+  container_name = "${lxd_container.container1.name}"
+  name = "snap1"
+}
+```
+
```
Comment (jtopjian) [inline README.md]: Can there be a reference with all parameters described?

File: lxd/resource_lxd_snapshot.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	snap, err := client.SnapshotInfo(snapId.LxdId())
+	if err != nil {
+		if err.Error() == "not found" {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	snapDump := spew.Sdump(snap)
+	log.Printf("[DEBUG] Snapshot Info: %s", snapDump)
```
Comment (jtopjian) [inline lxd/resource_lxd_snapshot.go]: Would `%#v` work just as well here? Spew is great for developing, but might be a bit too verbose in the logs afterwards.

Comment (sl1pm4t) [inline lxd/resource_lxd_snapshot.go]: Good catch, I meant to remove that.

File: lxd/resource_lxd_snapshot.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+	}
+
+	return false, err
+}
+
+type snapshotId struct {
+	remote    string
+	container string
+	snapshot  string
+}
```
Comment (jtopjian) [inline lxd/resource_lxd_snapshot.go]: I like this - nice!

**Comment by jtopjian** [comment]
Just two nits, but looks great :)

**Comment by jtopjian** [comment]
LGTM! I'll let you merge when ready 😄

**Pull request merged**